ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/filetype/DefaultExtensionToMimeTypeMappings.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.api.editor.filetype;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.inject.Singleton;
+
+/** @author Dmytro Kulieshov */
+@Singleton
+public class DefaultExtensionToMimeTypeMappings {
+  private final Map<String, Set<String>> mappings =
```

Comment:
We can make this static, it's immutable for the lifetime of the class. Lazy initialization/dependency injection gives no advantage here.

Reply:
>We can make this static, it's immutable for the lifetime of the class. Lazy initialization/dependency injection gives no advantage here.

Besides the fact that it preserves architectural homogeneity of the system and simplifies unit tests composition.